packages/ra-core/src/controller/usePaginationState.ts

```diff
@@ -53,7 +53,11 @@ export default (
     });
     const isFirstRender = useRef(true);
 
-    const setPerPage = useCallback(perPage => setPagination({ perPage }), []);
+    const setPerPage = useCallback(
+        perPage =>
+            setPagination({ perPage, page: initialPagination.page || 1 }),
```

Comment:
Why would you want to set the page to `initialPagination` on `perPage` change? I fail to see the rationale.